stream-chat-android-ui-common/src/main/kotlin/io/getstream/chat/android/ui/common/helper/DateFormatter.kt

```diff
@@ -92,8 +131,18 @@ internal class DefaultDateFormatter(
         return when {
             date.isToday() -> formatTime(date)
             date.isYesterday() -> dateContext.yesterdayString()
-            date.isWithinLastWeek() -> dateFormatterDayOfWeek.format(date)
-            else -> dateFormatterFullDate.format(date)
+            date.isWithinLastWeek() -> {
+                if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
+                    dateFormatterDayOfWeekNew.format(date.toInstant())
+                } else {
+                    dateFormatterDayOfWeek.format(date)
+                }
+            }
+            else -> if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
+                dateFormatterFullDateNew.format(date.toInstant())
+            } else {
+                dateFormatterFullDate.format(date)
+            }
```

Comment:
What about creating a couple of `DefaultDateFormater` depending on the SDK Version and only using the appropriate `DateFormater` removing unnecessary if/else blocks into the instance implementation?
At the end, the `DefaultDateFormater` entity is instantiated only within this file, the SDK Version check can be moved there.